File: code/frontend/fakemyresume/src/app/DTOs/ResumeDTO.ts
import { Education } from "./EducationDTO";
import { WorkExperience } from "./WorkExperienceDTO";

export class ResumeDTO {
    id?: number;
    accountId?: string;
    fullName!: string;
    currentRole!: string;
    description!: string;
    certifications!: string[];
    workExperience!: WorkExperience[];
    education!: Education[];

    constructor() {
        this.fullName = "";
        this.currentRole = "";
        this.description = "";
        this.certifications = [];
        this.workExperience = [];
        this.education = [];
    }
}

File: code/backend/FakeMyResume/DTOs/ResumeDTO.cs
using FakeMyResume.Models.Data;

namespace FakeMyResume.DTOs;

public class ResumeDTO : CreateResumeDTO
{
    public int Id { get; set; }

    public string AccountId { get; set; } = string.Empty;

    public string FullName { get; set; }
    public string Description { get; set; }
    public string CurrentRole { get; set; }

    public static ResumeDTO FromData(DataResume data)
    {
        var itm =new ResumeDTO();
        itm.Id = data.Id;
        itm.FullName = data.User.UserName;
        itm.Description = data.ResumeData.Description;
        itm.CurrentRole = data.ResumeData.CurrentRole;
        return itm;
    }
}
